packages/preview-middleware/src/base/flp.ts

```diff
@@ -273,20 +273,77 @@ export class FlpSandbox {
         this.router.use(PREVIEW_URL.client.path, serveStatic(PREVIEW_URL.client.local));
 
         // add route for the sandbox.html
-        this.router.get(this.config.path, (async (_req: Request, res: Response, next: NextFunction) => {
+        this.router.get(this.config.path, (async (
+            req: Request & { 'ui5-patched-router'?: { baseUrl?: string } },
+            res: Response,
+            next: NextFunction
+        ) => {
             // inform the user if a html file exists on the filesystem
             const file = await this.project.byPath(this.config.path);
             if (file) {
                 this.logger.info(`HTML file returned at ${this.config.path} is loaded from the file system.`);
                 next();
             } else {
-                const template = readFileSync(join(__dirname, '../../templates/flp/sandbox.html'), 'utf-8');
-                const html = render(template, this.templateConfig);
+                //ui5-patched-router comes with cds-plugin-ui5
+                const ui5Version = await this.getUi5Version(req.headers.host, req['ui5-patched-router']?.baseUrl ?? '');
```

Comment:
@petermuessig can I make use of the `req['ui5-patched-router'].baseUrl` property introduced by [cs-plugin-ui5](https://www.npmjs.com/package/cds-plugin-ui5)? Or is this something private/subject to change that I should rather not rely on?

Reply:
Checked with Peter, can be used.